### ISSUE TITLE
Probable moretrees planting solution

### ISSUE THREAD

**Issue by The4codeblocks** [sbrl/Minetest-WorldEditAdditions#104]
Use the "Fast Growth" variant of the saplings instead.

**Comment by sbrl** [comment]
Hello,

Thanks  for suggesting, and sorry I didn't see this issue earlier! Do you have the node name for the fast growth variant? I'm pretty sure I've tested it, but just want to make sure.

I've tested both the normal and the `_ongen` ones..... and neither of them work due to the fact that `moretrees` saplings don't work in the same way to other types of saplings, e.g. in `default`, `cooltrees`, etc.

**Comment by The4codeblocks** [comment]
(sorry for being late)
Is the problem that they're not being updated?

**Comment by sbrl** [comment]
The issue is that they are implemented differently and `moretrees` doesn't have an API that one can hook into to programmatically grow a tree IIRC @The4codeblocks.

We hook into TenPlus1's [bonemeal mod](https://content.minetest.net/packages/TenPlus1/bonemeal/) to do the growificating of trees, and while there is scope for implementing support for multiple bonemeal mods (considered for a separate purpose in #106), it would be much easier if `moretrees` supported being grown by `bonemeal` the same as every other sapling.

IIRC `moretrees` is community maintained over in <https://github.com/mt-mods/moretrees>, so a PR over there is needed to add an API we can hook into here.

**Comment by The4codeblocks** [comment]
I think what I meant is using the built-in generation by using the saplings themselves.

**Comment by sbrl** [comment]
I'm not sure what you mean by "built-in generation", @The4codeblocks?

**Comment by The4codeblocks** [comment]
> I'm not sure what you mean by "built-in generation", @The4codeblocks?

The moretrees generation used in saplings.

**Comment by sbrl** [comment]
Yeah, unfortunately the built-in generation function you mention is accessible to `moretrees` and not to other mods if I recall correctly.

**Comment by The4codeblocks** [comment]
> Yeah, unfortunately the built-in generation function you mention is accessible to `moretrees` and not to other mods if I recall correctly.

I'm trying to say that you _can_ call it, by placing a sapling.

**Comment by sbrl** [comment]
> I'm trying to say that you _can_ call it, by placing a sapling.

Right. Does that require a player to place the sapling, or can it be done by calling a function?

Currently WorldEditAdditions places saplings using [VoxelManipulators](https://api.minetest.net/lua-voxel-manipulator/), so it doesn't trigger any functions associated with nodes when they are placed.

I know there's also [`minetest.set_node()`](https://api.minetest.net/minetest-namespace-reference/#environment-access), but I haven't tried using this. It's also supposed to be quite inefficient if called a large number of times AFAIK.

**Comment by The4codeblocks** [comment]
well, you *can* use voxel manipulators if you also call the respective functions afterward

**Comment by sbrl** [comment]
You can do that, but IIRC the respective functions don't immediately grow the tree? If you've got a link to their API I'd really appreciate it
